### PR TITLE
Add core files.

### PR DESCRIPTION
**Pull request by mattcollier**
There is a rather serious fly in the ointment here!

Due to the error handling features in ky, the status code from a 404 error is being lost as far as I can tell.

Fetch does not throw on a 404 and if it reports a status code, ky is not including that in the error it is throwing.  The test suite here demonstrates this behavior.  Maybe I'm missing something.  Going to investigate further.

TODO:
- [ ] Use mime lib for response detection [EDIT: not doing this now]

### CODE REVIEW

**Comment by mattcollier** [comment]
Ready to go here, I've implemented this in edv-client and webkms-client libraries and tested in some dependent modules and it's working well.

File: main.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      return response;
+    };
+  }
+});
+
+async function _handleError(e) {
+  // handle network errors that do not have a response
+  if(!e.response) {
+    if(e.message === 'Failed to fetch') {
+      e.message = `${e.message}. Possible CORS error.`;
```
Comment (davidlehn) [inline main.js:45]: Should adding the CORS message be run just on browsers?

Comment (mattcollier) [reply main.js:45]: @davidlehn as far as we know this 'Failed to fetch' message is only generated in browsers.

File: tests/10-client-api.spec.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    const {httpClient, ky} = dbHttpClient;
+    httpClient.should.be.a('function');
+    ky.should.be.a('function');
+  });
+  it('handles a network error', async () => {
+    const {httpClient} = dbHttpClient;
+    let err;
+    let response;
+    try {
+      // hit a non-existent endpoint on the karma server to avoid CORS concerns
```
Comment (davidlehn) [inline tests/10-client-api.spec.js:22]: Not quite accurate when run on node.  Get ECONNREFUSED as there is no server.  Is that ok?  Also ok to assume no server happens to be running there?

File: tests/10-client-api.spec.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    } else { // failed to fetch may commonly be due to an issue with CORS
+      err.message.should.equal('Failed to fetch. Possible CORS error.');
+    }
+  });
+  it('succesfully makes request with default json headers', async () => {
+    const {httpClient} = dbHttpClient;
+    let err;
+    let response;
+    try {
+      // TODO: use another site/mock for error response
```
Comment (davidlehn) [inline tests/10-client-api.spec.js:79]: How important is it to fix these with a local server?  Should we make an issue for it or address it before first release?  httpbin sure is easier!

File: README.md [hunk 4]
```diff
@@ -1,2 +1,68 @@
...
+```
+
+#### GET a JSON response in the browser
+```js
+try {
+  result = await httpClient.get('http://httpbin.org/json');
+  return result.data;
+} catch(e) {
+  // status is HTTP status code
+  // data is JSON error from the server
```
Comment (davidlehn) [inline README.md:18]: A bit tricky to follow the error path, but if you ask for json, and got back a 404 with html content, data would just be undefined right?  Maybe this should say "data is JSON error from the server if available"?

File: tests/10-client-api.spec.js (hunk 3 above)
Comment (mattcollier) [reply tests/10-client-api.spec.js:79]: httbin is awesome, we're rolling with it, removing the todo.

**Pull request merged**